demoscene/utils/groklinks.py

```diff
@@ -1531,12 +1531,70 @@ def __unicode__(self):
         return u"https://%s.bandcamp.com/track/%s" % (domain, name)
 
 
+class Defactro2Entry(BaseUrl): # for use as an abstract superclass
+    html_link_class = "defacto2"
+    html_link_text = "Defacto2"
+    html_title_format = "%s on Defacto2"
+
+
+class Defacto2File(Defactro2Entry):
+    canonical_format = "https://defacto2.net/f/%s"
+    tests = [
+        # XXX: See if this works with DOSee config :/
+        regex_match(r'https?://defacto2\.net/f/([\w-]+)', re.I),
+        regex_match(r'https?://defacto2\.net/file/detail/([\w-]+)', re.I),
+    ]
+
+
+class Defacto2Person(Defactro2Entry):
+    canonical_format = "https://defacto2.net/p/%s"
+    tests = [
+        regex_match(r'https?://defacto2\.net/p/([\w-]+)', re.I),
+    ]
+
+
+class Defacto2Group(Defactro2Entry):
+    canonical_format = "https://defacto2.net/g/%s"
+    tests = [
+        regex_match(r'https?://defacto2\.net/g/([\w-]+)', re.I),
+    ]
+
+
+class RetroSceneEntry(BaseUrl): # for use as an abstract superclass
+    html_link_class = "retroscene"
+    html_link_text = "RetroScene"
+    html_title_format = "%s on RetroScene"
+
+
+class RetroSceneEvent(RetroSceneEntry):
+    canonical_format = "https://events.retroscene.org/%s"
+    tests = [
+        regex_match(r'https?://events\.retroscene\.org/([\w-]+)', re.I),
+    ]
+
+
+class RetroSceneRelease(RetroSceneEntry):
+    def match_retroscene_release(urlstring, url):
+        regex = re.compile(r'https?://events\.retroscene\.org/([\w-]+)/([\w-]+)/([\w-]+)', re.I),
+        match = regex.match(urlstrings)
+        if match:
+            event, compoName, entryNumber = match.groups()
+            return "%s/%s/%s" % (event, compoName, entryNumber)
+        
+        tests = [match_retroscene_release]
```

Comment:
This (and the code below) is indented too much - it's inside the `match_retroscene_release` method, which means it never replaces the `tests` from BaseUrl and ends up matching all links.